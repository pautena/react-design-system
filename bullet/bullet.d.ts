import { SxProps, Theme } from '@mui/material/styles';
export type BulletVariant = "primary" | "secondary" | "default" | "info" | "warning" | "error";
export declare const bulletClasses: {
    root: string;
};
export interface BulletProps {
    /**
     * Color palette used to draw the component
     */
    variant?: BulletVariant;
    /**
     * Custom styles
     */
    sx?: SxProps<Theme>;
}
/**
 * Dot to attract the user attention
 */
export declare const Bullet: ({ variant, sx }: BulletProps) => import("react/jsx-runtime").JSX.Element;
