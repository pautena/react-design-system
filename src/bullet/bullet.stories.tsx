import type { Meta, StoryObj } from "@storybook/react";
import { Bullet } from "./bullet";

export default {
	title: "Components/Data Display/Bullet",
	component: Bullet,
	parameters: {
		layout: "centered",
	},
} satisfies Meta<typeof Bullet>;
type Story = StoryObj<typeof Bullet>;

export const Default: Story = {
	args: {
		variant: "primary",
	},
};

export const PrimaryVariant: Story = {
	args: {
		variant: "primary",
	},
};

export const SecondaryVariant: Story = {
	args: {
		variant: "secondary",
	},
};

export const DefaultVariant: Story = {
	args: {
		variant: "default",
	},
};

export const InfoVariant: Story = {
	args: {
		variant: "info",
	},
};

export const WarningVariant: Story = {
	args: {
		variant: "warning",
	},
};

export const ErrorVariant: Story = {
	args: {
		variant: "error",
	},
};
