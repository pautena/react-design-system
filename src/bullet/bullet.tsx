import { Badge, SxProps, Theme } from "@mui/material";
import React from "react";

export type BulletVariant = "primary" | "secondary" | "default" | "info" | "warning" | "error";

export const bulletClasses = {
  root: "RdsBullet-root",
};

export interface BulletProps {
  /**
   * Color palette used to draw the component
   */
  variant?: BulletVariant;
  /**
   * Custom styles
   */
  sx?: SxProps<Theme>;
}

/**
 * Dot to attract the user attention
 */
export const Bullet = ({ variant = "primary", sx }: BulletProps) => {
  return (
    <Badge
      color={variant}
      variant="dot"
      className={bulletClasses.root}
      role="bullet"
      aria-describedby={variant}
      sx={sx}
    />
  );
};
