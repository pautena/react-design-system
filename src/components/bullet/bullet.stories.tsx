import { ComponentMeta } from "@storybook/react";
import { createTemplate } from "../../storybook";
import { Bullet } from "./bullet";

export default {
  title: "Components/Bullet",
  component: Bullet,
  parameters: {
    layout: "centered",
  },
} as ComponentMeta<typeof Bullet>;

const Template = createTemplate(Bullet);

export const Default = Template.bind({});
Default.args = {
  variant:'default'
};

export const Primary = Template.bind({});
Primary.args = {
  variant:'primary'
};

export const Secondary = Template.bind({});
Secondary.args = {
  variant:'secondary'
};

export const Info = Template.bind({});
Info.args = {
  variant:'info'
};

export const Warning = Template.bind({});
Warning.args = {
  variant:'warning'
};

export const Error = Template.bind({});
Error.args = {
  variant:'error'
};