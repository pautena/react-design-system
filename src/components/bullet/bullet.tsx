import { Badge } from '@mui/material';
import React from 'react';

export type BulletVariant = "primary" | "secondary" | "default" | "info" | "warning" | "error";

export interface BulletProps {
  variant?: BulletVariant;
}


export const Bullet = ({variant='primary'}:BulletProps)=> {
  return <Badge color={variant} variant="dot" />
}