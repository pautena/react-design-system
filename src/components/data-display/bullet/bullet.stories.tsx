import { Meta, StoryObj } from "@storybook/react";
import { Bullet } from "./bullet";

export default {
  title: "Components/Data Display/Bullet",
  component: Bullet,
  parameters: {
    layout: "centered",
  },
} satisfies Meta<typeof Bullet>;
type Story = StoryObj<typeof Bullet>;

export const Primary: Story = {
  args: {
    variant: "primary",
  },
};

export const Secondary: Story = {
  args: {
    variant: "secondary",
  },
};

export const Default: Story = {
  args: {
    variant: "default",
  },
};

export const Info: Story = {
  args: {
    variant: "info",
  },
};

export const Warning: Story = {
  args: {
    variant: "warning",
  },
};

export const Error: Story = {
  args: {
    variant: "error",
  },
};
