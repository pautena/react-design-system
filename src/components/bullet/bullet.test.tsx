import React from "react";
import { Bullet, BulletVariant } from "./bullet";
import { render, screen } from "../../tests";

describe("Bullet", () => {
  const renderComponent = (variant: BulletVariant | undefined = undefined) => {
    return render(<Bullet variant={variant} />);
  };

  it("renders as primary without a variant", () => {
    renderComponent(undefined);

    expect(screen.getByRole("bullet")).toHaveAttribute("aria-describedby", "primary");
  });

  it.each([["primary"], ["secondary"], ["default"], ["info"], ["warning"], ["error"]])(
    "renders correctly with variant %s",
    (variant: string) => {
      const { container } = renderComponent(variant as BulletVariant);

      expect(container).toMatchSnapshot();
      expect(screen.getByRole("bullet")).toHaveAttribute("aria-describedby", variant);
    },
  );
});
