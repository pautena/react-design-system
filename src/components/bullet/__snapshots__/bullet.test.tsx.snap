// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bullet renders correctly with variant default 1`] = `
<div>
  <span
    aria-describedby="default"
    class="RdsBullet-root MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
    role="bullet"
  >
    <span
      class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular BaseBadge-badge css-qc5jhc-MuiBadge-badge"
    />
  </span>
</div>
`;

exports[`Bullet renders correctly with variant error 1`] = `
<div>
  <span
    aria-describedby="error"
    class="RdsBullet-root MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
    role="bullet"
  >
    <span
      class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError BaseBadge-badge css-hqv4f7-MuiBadge-badge"
    />
  </span>
</div>
`;

exports[`Bullet renders correctly with variant info 1`] = `
<div>
  <span
    aria-describedby="info"
    class="RdsBullet-root MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
    role="bullet"
  >
    <span
      class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorInfo BaseBadge-badge css-17bmh8a-MuiBadge-badge"
    />
  </span>
</div>
`;

exports[`Bullet renders correctly with variant primary 1`] = `
<div>
  <span
    aria-describedby="primary"
    class="RdsBullet-root MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
    role="bullet"
  >
    <span
      class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary BaseBadge-badge css-vzhiio-MuiBadge-badge"
    />
  </span>
</div>
`;

exports[`Bullet renders correctly with variant secondary 1`] = `
<div>
  <span
    aria-describedby="secondary"
    class="RdsBullet-root MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
    role="bullet"
  >
    <span
      class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary BaseBadge-badge css-2liczv-MuiBadge-badge"
    />
  </span>
</div>
`;

exports[`Bullet renders correctly with variant warning 1`] = `
<div>
  <span
    aria-describedby="warning"
    class="RdsBullet-root MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
    role="bullet"
  >
    <span
      class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorWarning BaseBadge-badge css-120limi-MuiBadge-badge"
    />
  </span>
</div>
`;
